libs/tree-sitter-wing/grammar.js

```diff
@@ -538,6 +539,20 @@ module.exports = grammar({
       seq($.identifier, ":", $.expression),
     structured_access_expression: ($) =>
       prec.right(seq($.expression, "[", $.expression, "]")),
+
+    json_literal: ($) => seq(field("type", $.json_container_type), field("element", $.json_element)),
+    
+    json_element: ($) => choice(
+      $._literal,
+      $.map_literal,
+      $.array_literal
```

Comment:
There are two things that `Json` literals have that "normal" literals don't which we should make sure we take care of:
1) I think json maps accept both `key: val` and `"key": val` format. This means that we need to change the grammar for `map_literal` to support the string variant (or is it even an `$.expression` variant)? Later on during the parsing phase we can report and error if someone tries to use it out of place.
2) `array_literal`s need to be uniformly typed when not a `Json` array. This needs to be verified in the type checker.

**1** above might introduce a world of complexity if we support map keys as expressions, so it might be worth starting without this, just string literals or simple identifiers.

Reply:
For 1 it looks like we already do support both formats with normal map_literals

2. Yes agree, this has been using RefCell booleans